ModelBundle/Migrations/MongoDB/Version20160601155518.php

```diff
@@ -0,0 +1,283 @@
+                            keyword = {};
+                            keyword._id = keywords[i].$id
+                            db.keyword.find({"_id" : keywords[i].$id}).forEach(function(record) {
+                                for (property in record) {
+                                    keyword[property] = record[property];
+                                }
+                            });
+                            item.keywords.push(keyword);
+                        }
+                        db.' . $collection . '.update({ _id: item._id }, item);
+                    }
+                });
+            ');
+        }
+
+        $this->parseNode('down');
+    }
+
+    /**
+     * Load keywordable collection
+     */
+    protected function getKeywordableCollection() {
+        $this->collections = array();
+        $metadataFactory = $this->container->get('object_manager')->getMetadataFactory();
+        $allMetadata = $metadataFactory->getAllMetadata();
+
+        foreach ($allMetadata as $metadata) {
+            try {
+                $document = $metadata->newInstance();
+                if (array_key_exists('OpenOrchestra\MongoTrait\Keywordable', class_uses($document))) {
+                    $this->collections[] = $metadata->getCollection();
+                }
+            } catch (\Exception $e) {}
+        }
+    }
+
+    /**
+     * Load the configuration file
+     */
+    protected function loadConfiguration()
+    {
+        $yamlParser = new YamlParser();
+
+        $configurationFilePath =
+            $this->container->getParameter('kernel.root_dir') . '/config/keyword_migration.yml';
+
+        if (is_file($configurationFilePath)) {
+            $this->configuration = $yamlParser->parse(file_get_contents($configurationFilePath));
+        } else {
+            $migrationDir = $this->container->getParameter('mongo_db_migrations.dir_name');
+            $this->configuration = $yamlParser->parse(file_get_contents($migrationDir . '/config/keyword_migration.yml'));
+        }
+        if (!array_key_exists('condition_csv', $this->configuration)) {
+            $this->configuration['condition_csv'] = array();
+        }
+        if (!array_key_exists('condition_boolean', $this->configuration)) {
+            $this->configuration['condition_boolean'] = array();
+        }
+    }
+
+    protected function parseNode($direction) {
+        $step = 10;
+        $documentManager = $this->container->get('object_manager');
+        $countNodes = $documentManager->createQueryBuilder('OpenOrchestra\ModelBundle\Document\Node')
+            ->getQuery()
+            ->execute()
+            ->count();
+        for ($i = 0; $i < intval($countNodes / $step) + 1; $i++) {
+            $nodes = $documentManager->createQueryBuilder('OpenOrchestra\ModelBundle\Document\Node')
+                ->limit($step)
+                ->skip($i * $step)
+                ->getQuery()
+                ->execute();
```

Comment:
you should add an order tag or a criterion based on the fact that the transformation has been made or not, because you have no warranty that a node already altered is not returned in a next loop